specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/secrets.json

```diff
@@ -572,6 +572,137 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The secret rotation policy.",
+            "schema": {
+              "$ref": "#/definitions/SecretRotationPolicy"
+            }
+          },
+          "default": {
+            "description": "Key Vault error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/KeyVaultError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetKeyRotationPolicy": {
+            "$ref": "./examples/GetSecretRotationPolicy-example.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Secrets"
+        ],
+        "operationId": "UpdateSecretRotationPolicy",
+        "summary": "Updates the rotation policy for a secret.",
+        "description": "Set rotation policy for a secret. Leave others as undefined. This operation requires the secret rotation policy set permission.",
+        "parameters": [
+          {
+            "name": "secret-name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the secret in the given vault."
+          },
+          {
+            "name": "secretRotationPolicy",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SecretRotationPolicy"
+            },
+            "description": "The rotation policy for the secret."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The rotation policy for the secret.",
+            "schema": {
+              "$ref": "#/definitions/SecretRotationPolicy"
+            }
+          },
+          "default": {
+            "description": "Key Vault error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/KeyVaultError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "UpdateKeyRotationPolicy": {
+            "$ref": "./examples/UpdateSecretRotationPolicy-example.json"
+          }
+        }
+      }
+    },
+    "/secrets/{secret-name}/rotate": {
+      "post": {
+        "tags": [
+          "Secrets"
+        ],
+        "operationId": "RotateSecret",
+        "summary": "Creates a new secret version, stores it, then returns secret value, attributes to the client.",
+        "description": "The operation will rotate the secret based on the secret rotation policy. It requires the secrets/rotate permission.",
+        "parameters": [
+          {
+            "name": "secret-name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[0-9a-zA-Z-]+$",
```

Comment:
Why is there a pattern here but nowhere else? Also, `secret-name` is repeated enough that maybe it should be in a `"parameters"` section and just `ref` it again and again. Just a thought, though.

Reply:
The pattern exists in secret create api. I copied the parameter from there. I have moved it to a separated parameter and ref it in places where it was using